### PR TITLE
Added Program hash changed event

### PR DESCRIPTION
**Pull request by Akashneelesh**
- [x] issue #6 
- [ ] follows contribution [guide](../CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

Implemented the event (ProgramHashChanged) for the function set_program_info

### CODE REVIEW

**Comment by Akashneelesh** [comment]
> hey man, thanks for that! Stacked on #12 right?

Yess ser

File: src/config/component.cairo [hunk 1]
```diff
@@ -51,8 +64,16 @@ mod config_cpt {
...
-            self.program_info.write((program_hash, config_hash))
-        // TODO(#6): ProgramHashChanged Event
+            let (old_program_hash, _): (felt252, felt252) = self.program_info.read();
+            self.program_info.write((program_hash, config_hash));
+            self
+                .emit(
+                    ProgramHashChanged {
+                        changed_by: get_caller_address(),
+                        old_program_hash: old_program_hash,
+                        new_program_hash: program_hash,
```
[the remark this answers is not in the document]
Comment (Akashneelesh) [reply src/config/component.cairo:74]: Yep true, it will have both the program_hash and config_hash

File: src/config/component.cairo [hunk 2]
```diff
@@ -18,7 +18,7 @@ mod config_cpt {
         interface::IOwnable,
     };
     use piltover::config::interface::IConfig;
-    use starknet::ContractAddress;
+    use starknet::{ContractAddress, get_caller_address};
```
Comment (b-j-roberts) [inline src/config/component.cairo:21]: Our current style is to not import functions like this, instead when you use the function do `Starknet::get_caller_address`

File: src/config/component.cairo (hunk 1 above)
[the remark this answers is not in the document]
Comment (b-j-roberts) [reply src/config/component.cairo:74]: I'd say either create a `ProgramInfoChanged` event or add the `ConfigHashChanged` event like Solidity and emit both events instead. Up to you.

File: src/config/component.cairo (hunk 2 above)
Comment (Akashneelesh) [reply src/config/component.cairo:21]: Got it, I'll do the change now

File: src/config/component.cairo (hunk 1 above)
[the remark this answers is not in the document]
Comment (Akashneelesh) [reply src/config/component.cairo:74]: Okay I'll just change it into ProgramInfoChanged and commit it now

**Review by b-j-roberts — APPROVED** [review]

**Pull request merged**